utils/tracking.ts

```diff
@@ -0,0 +1,90 @@
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { EventTelemetry } from "applicationinsights/out/Declarations/Contracts";
+import { UpdateSubscriptionFeedInput } from "../UpsertServicePreferences/subscription_feed";
+import { initTelemetryClient } from "./appinsights";
+import { toHash } from "./crypto";
+
+export const createTracker = (
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+) => {
+  const eventName = (name: string) => `api.profile.${name}`;
+
+  /**
+   * Trace an event when a user changes their preference mode
+   */
+  const traceServicePreferenceModeChange = (
+    hashedFiscalCode: string,
+    previousMode: ServicesPreferencesModeEnum,
+    nextMode: ServicesPreferencesModeEnum
```

Comment:
Could be useful add the current profile version

Reply:
done in https://github.com/pagopa/io-functions-app/pull/185/commits/5d030d97a8a541647b3860511958bc8460f9a78b